src/libopensc/card-openpgp.c

```diff
@@ -1856,6 +1894,42 @@ pgp_select_data(sc_card_t *card, u8 p1)
 	LOG_FUNC_RETURN(card->ctx, r);
 }
 
+/**
+ * Internal: Same as pgp_select_data, but with Yubikey workaround
+ * https://github.com/Yubico/yubikey-manager/issues/403
+ *
+ * p1: number of an instance (DO 7F21: 0x00 for AUT, 0x01 for DEC and 0x02 for SIG)
+ */
+static int
+pgp_select_data_yubikey(sc_card_t *card, u8 p1)
+{
+	sc_apdu_t	apdu;
+	u8	apdu_data[7];
+	int	r;
+
+	LOG_FUNC_CALLED(card->ctx);
+
+	sc_log(card->ctx, "select data with: %u (Yubikey workaround)", p1);
+	// Yubikey wants another length
+	apdu_data[0] = 0x06;
```

Comment:
given that the code change is basically just this line, would it make sense to have the check for the flag checked directly in `pgp_select_data()` instead of duplicating the whole function? it will need a slight change in how the data is constructed to something like
```
if (card->flags & SC_CARD_FLAG_YUBIKEY_SELECT) { 
	apdu_data[n++] = 0x06;
}
// the rest of assgined in format of apdu_data[n++] = 0x06;
```
and then use `n` instead of `sizeof(adu_data)` below?